src/common/dns_utils.cpp

```diff
@@ -207,14 +209,31 @@ DNSResolver::DNSResolver() : m_data(new DNSResolverData())
       LOG_PRINT_L0("Using public DNS server: " << dns_public_addr << " (TCP)");
       use_dns_public = 1;
     }
+    else if (sscanf(dns_public.c_str(), "tcp://%u.%u.%u.%u%c", &ip0, &ip1, &ip2, &ip3, &c) == 4)
```

Comment:
Why this additional `char c`? I tested it and it works also without it.

Reply:
Try tcp://1.2.3.4x with and without.

Reply:
Ah I see it detects it as invalid. :)

Reply:
Is it worth adding a test case so that if someone removes the `%c` the test fails?

Reply:
done